File: pubnub-gson/pubnub-gson-impl/src/main/java/com/pubnub/internal/endpoints/objects_api/uuid/RemoveUUIDMetadataImpl.java
package com.pubnub.internal.endpoints.objects_api.uuid;

import com.pubnub.api.endpoints.objects_api.uuid.RemoveUUIDMetadata;
import com.pubnub.api.endpoints.remoteaction.ExtendedRemoteAction;
import com.pubnub.api.endpoints.remoteaction.MappingRemoteAction;
import com.pubnub.api.models.consumer.objects_api.uuid.PNRemoveUUIDMetadataResult;
import com.pubnub.internal.EndpointInterface;
import com.pubnub.internal.PubNubCore;
import com.pubnub.internal.endpoints.DelegatingEndpoint;
import com.pubnub.internal.models.consumer.objects.PNRemoveMetadataResult;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.jetbrains.annotations.NotNull;

@Setter
@Accessors(chain = true, fluent = true)
public class RemoveUUIDMetadataImpl extends DelegatingEndpoint<PNRemoveMetadataResult, PNRemoveUUIDMetadataResult> implements RemoveUUIDMetadata {

    private String uuid;

    public RemoveUUIDMetadataImpl(final PubNubCore pubnub) {
        super(pubnub);
    }

    @Override
    @NotNull
    protected EndpointInterface<PNRemoveMetadataResult> createAction() {
        return pubnub.removeUUIDMetadata(uuid);
    }

    @NotNull
    @Override
    protected ExtendedRemoteAction<PNRemoveUUIDMetadataResult> mapResult(@NotNull ExtendedRemoteAction<PNRemoveMetadataResult> action) {
        return new MappingRemoteAction<>(
                action,
                result -> new PNRemoveUUIDMetadataResult(
                        result.getStatus(),
                        null
                )
        );
    }
}

File: pubnub-kotlin/src/main/kotlin/com/pubnub/internal/kotlin/endpoints/objects/uuid/RemoveUUIDMetadataImpl.kt
package com.pubnub.internal.kotlin.endpoints.objects.uuid

import com.pubnub.api.DelegatingEndpoint
import com.pubnub.api.endpoints.remoteaction.ExtendedRemoteAction
import com.pubnub.api.endpoints.remoteaction.MappingRemoteAction
import com.pubnub.api.models.consumer.objects.PNRemoveMetadataResult
import com.pubnub.internal.endpoints.objects.uuid.IRemoveUUIDMetadata
import com.pubnub.internal.endpoints.objects.uuid.RemoveUUIDMetadata

class RemoveUUIDMetadataImpl internal constructor(removeUUIDMetadata: RemoveUUIDMetadata) :
    DelegatingEndpoint<PNRemoveMetadataResult, com.pubnub.internal.models.consumer.objects.PNRemoveMetadataResult>(
        removeUUIDMetadata
    ),
    IRemoveUUIDMetadata by removeUUIDMetadata,
    com.pubnub.api.endpoints.objects.uuid.RemoveUUIDMetadata {
    override fun convertAction(remoteAction: ExtendedRemoteAction<com.pubnub.internal.models.consumer.objects.PNRemoveMetadataResult>): ExtendedRemoteAction<PNRemoveMetadataResult> {
        return MappingRemoteAction(remoteAction, PNRemoveMetadataResult.Companion::from)
    }
}
